tools/serve/serve.py

```diff
@@ -827,12 +829,15 @@ def run(**kwargs):
             servers = start(config, build_routes(config["aliases"]), **kwargs)
 
             try:
-                while any(item.is_alive() for item in iter_procs(servers)):
+                while all(item.is_alive() for item in iter_procs(servers)):
                     for item in iter_procs(servers):
                         item.join(1)
             except KeyboardInterrupt:
                 logger.info("Shutting down")
 
+        for item in iter_procs(servers):
```

Comment:
A sample of the output I see:
```
^CINFO:web-platform-tests:Shutting down
INFO:web-platform-tests:Status of ws on port 61486:	running
INFO:web-platform-tests:Status of wss on port 61487:	running
INFO:web-platform-tests:Status of http on port 8000:	not running
INFO:web-platform-tests:Status of http on port 61488:	not running
INFO:web-platform-tests:Status of https on port 8443:	not running
```

As a user of `./wpt serve` manually, I'm not sure what to do with this on shutdown, if it's cause for concern or not. Do you think these make more sense as debug messages?

Alternatively, if the reason we shut down is that some subprocess has died, can we log that fact clearly, and just in that case?

Reply:
I prefer the second alternative because this information does seem relevant to the typical user of `wpt serve`, just not so much in the "Ctrl-C" case. Here's what it looks like now (printed only in response to subprocess exit):

>     INFO:web-platform-tests:2 subprocesses exited:
>     INFO:web-platform-tests:Status of ws on port 44283:     running
>     INFO:web-platform-tests:Status of wss on port 44819:    running
>     INFO:web-platform-tests:Status of http on port 8000:    not running
>     INFO:web-platform-tests:Status of http on port 40161:   running
>     INFO:web-platform-tests:Status of https on port 8443:   not running